traversal/planner/GraphPlanner.java

```diff
@@ -410,47 +386,35 @@ public String toString() {
         return str.toString();
     }
 
-    private class Initialiser {
-
-        private final LinkedHashSet<PlannerVertex<?>> queue;
-        private int edgeCount;
-
-        private Initialiser() {
-            queue = new LinkedHashSet<>();
-            edgeCount = 0;
+    private void setInitialValues() {
+        resetInitialValues();
+
+        PlannerVertex<?> start = vertices.values().stream().min(comparing(PlannerVertex::cost)).get();
+        Set<PlannerVertex<?>> closedSet = new HashSet<>();
+        PriorityQueue<PlannerVertex<?>> open = new PriorityQueue<>(comparing(
+                v -> v.ins().stream().filter(e -> closedSet.contains(e.from())).mapToDouble(PlannerEdge.Directional::cost).min().orElse(v.cost())
```

Comment:
As far as I can tell, `PriorityQueue` doesn't recalculate the priority once the element was inserted.
I don't think this is that big of a problem, we just want the best value we can get at the time we add the vertex to the queue. Just using the vertex's `cost()` is not great, because we will end up just sorting the vertices without considering the edges at all. This could be rewritten without using a priority queue at all, and I think I just will.

Reply:
Just responding to: "As far as I can tell, PriorityQueue doesn't recalculate the priority once the element was inserted."

That may be the case in the specific implementation used, but we shouldn't generally assume how a class is implementing its internal state. Additionally, that assumption isn't actually in line with my expection of how a priority queue might be implemented (when adding new elements to the queue, you have to re-evaluate the comparator to place elements correctly).

The refactored version without the `PriorityQueue` looks good!!